tools/autograd/gen_variable_type.py

```diff
@@ -85,7 +85,7 @@
 
 DERIVATIVE_MULTI = CodeTemplate("""\
 if (should_compute_output({ ${idxs} })) {
-  auto output_mask = std::array<bool, ${n}>{
+  auto ${output_mask} = std::array<bool, ${n}>{
```

Comment:
Yeah, we can leave it as `output_mask`, if the "other" mask is never useful. However, what we should do is error if someone tries to name an input parameter `output_mask`, since it will get shadowed by the `output_mask` we autogenerate.

Reply:
I ended up renaming this to 'grad_input_mask', but doing so uniformly for everything.